### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by Koliras**
- [DEMO LINK](https://Koliras.github.io/layout_calendar/)
- [TEST REPORT LINK](https://Koliras.github.io/layout_calendar/report/html_report/)

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -7,6 +7,68 @@
...
   <body>
-    <h1>Calendar</h1>
+    <div class="calendar calendar--month-length--31 calendar--start-day--sun">
+      <div class="calendar__day"></div>
+      
+      <div class="calendar__day"></div>
+      
+      <div class="calendar__day"></div>
+      
+      <div class="calendar__day"></div>
```
Comment (alexander-ignatow) [inline src/index.html]: It is better to use semantic tags for calendar and cells

File: src/styles/main.scss [hunk 2]
```diff
@@ -1,3 +1,70 @@
...
+  display: flex;
+  justify-content: center;
+  align-items: center;
   margin: 0;
 }
+
+.calendar {
+  display: flex;
+  justify-content: flex-start;
+  // align-items: ;
```
Comment (alexander-ignatow) [inline src/styles/main.scss]: Avoid keeping commented out  code

File: src/styles/main.scss [hunk 3]
```diff
@@ -1,3 +1,70 @@
...
+$monthLength: 28, 29, 30, 31;
+$startDay: (
+  "mon": 0,
+  "tue": ($cellSize + $gap),
+  "wed": 2 * ($cellSize + $gap),
+  "thu": 3 * ($cellSize + $gap),
+  "fri": 4 * ($cellSize + $gap),
+  "sat": 5 * ($cellSize + $gap),
+  "sun": 6 * ($cellSize + $gap)
+);
```
Comment (alexander-ignatow) [inline src/styles/main.scss]: *grammar nazi*

for collections use plural variable names:
monthLength -> monthLengths
etc

**Review by alexander-ignatow — CHANGES_REQUESTED** [review]
Very good!

Just few touches to improve it

File: src/index.html [hunk 4]
```diff
@@ -7,6 +7,68 @@
     <link rel="stylesheet" href="./styles/main.scss">
   </head>
   <body>
-    <h1>Calendar</h1>
+    <ul class="calendar calendar--month-length--31 calendar--start-day--sun">
+      <li class="calendar__day"></li>
+
+      <li class="calendar__day"></li>
+
+      <li class="calendar__day"></li>
```
Comment (tiserett) [inline src/index.html]: I prefer not to add gaps between such containers, those tags are the same, I don't see a reason to add a gap between them

File: src/styles/main.scss [hunk 5]
```diff
@@ -1,3 +1,70 @@
...
+$monthLengths: 28, 29, 30, 31;
+$startDays: (
+  "mon": 0,
+  "tue": ($cellSize + $gap),
+  "wed": 2 * ($cellSize + $gap),
+  "thu": 3 * ($cellSize + $gap),
+  "fri": 4 * ($cellSize + $gap),
+  "sat": 5 * ($cellSize + $gap),
+  "sun": 6 * ($cellSize + $gap)
+);
```
Comment (tiserett) [inline src/styles/main.scss]: separate file for variables please

File: src/styles/main.scss [hunk 6]
```diff
@@ -1,3 +1,70 @@
+
+$cellSize: 100px;
+$gap: 1px;
+$calendarSize: calc(#{$cellSize} * 7 + #{$gap} * 6);
```
Comment (tiserett) [inline src/styles/main.scss]: I guess you can avoid using calc with scss, am I wrong?

btw, what is 7 and 6?

**Review by tiserett — CHANGES_REQUESTED** [review]
try to center calendar vertically as well

File: src/styles/main.scss (hunk 6 above)
Comment (Koliras) [reply src/styles/main.scss]: I tried without calc but it didn't work, but I'll try once more.

7 is the amount of days in a row and 6 is the amount of gaps between them

**Review by BudnikOleksii — APPROVED** [review]
👍🏻

**Pull request closed without merging**